pontoon/teams/views.py

```diff
@@ -379,6 +385,143 @@ def ajax_translation_memory_delete(request, locale):
     return HttpResponse("ok")
 
 
+@require_AJAX
+@require_POST
+@permission_required_or_403("base.can_translate_locale", (Locale, "code", "locale"))
+@transaction.atomic
+def ajax_translation_memory_upload(request, locale):
+    """Upload Translation Memory entries from a .TMX file."""
+    try:
+        file = request.FILES["tmx_file"]
+    except MultiValueDictKeyError:
+        return JsonResponse(
+            {"status": False, "message": "No file uploaded."},
+            status=400,
+        )
+
+    if not file.name.endswith(".tmx"):
+        return JsonResponse(
+            {
+                "status": False,
+                "message": "Invalid file format. Only .TMX files are supported.",
+            },
+            status=400,
+        )
+
+    locale = get_object_or_404(Locale, code=locale)
+    code = locale.code
+
+    # Parse the TMX file
+    try:
+        tree = ET.parse(file)
+        root = tree.getroot()
+    except ET.ParseError as e:
+        return JsonResponse(
+            {"status": False, "message": f"Invalid XML file: {e}"}, status=400
+        )
+
+    # Extract TM entries
+    file_entries = []
+    srclang_pattern = re.compile(r"^en([-_].+)?$", re.IGNORECASE)
```

Comment:
This feels a bit overkill.

Do we care about excluding English variants from this? That was my understanding: we want `en` and `en-US`, the latter with possible weird spelling, not `en-GB` and `en-CA`.

In case, the group should probably be non-capturing (`^en(?:[-_].+)?$`).

Reply:
We don't do that when it [comes to VCS](https://github.com/mozilla/pontoon/blob/bc2f3642a93d797553f662720ed86e68cf0b00bf/pontoon/sync/vcs/project.py#L50-L57), why would we want a different behavior for TM import?

Reply:
Note that with the sync refactor, the VCS source discovery moves to moz.l10n and becomes [configurable](https://github.com/mozilla/moz-l10n/blob/615ee50f034203bd9fe9a853e98804fcd8995dd8/moz/l10n/paths/discover.py#L83-L86).

But that's not really relevent, tbh. The cases are different because the project paths are controlled by admins, while the TMX values are controlled by individual translators. The former is set when the project is initially configured, while the latter can happen at any time by ~anyone.

But my thoughts here are not very strong, esp. as this parser is getting inlined within the view code like this.